feat: add `How Secrets Sprawl is Slowing You Down—And What to Do About It` blog (#13095)
pulumi/docs: content/blog/how-secrets-sprawl-is-slowing-you-down/index.md

---
title: "How Secrets Sprawl is Slowing You Down—And What to Do About It."
allow_long_title: true
date: 2024-10-10T12:40:10+02:00
draft: false

meta_desc: "Secrets sprawl is a real problem that can slow down our teams and impact our productivity. But it also affects how fast an organization can scale its development and operations teams. Learn how to overcome secrets sprawl and regain your productivity."

meta_image: meta.png

authors:
- engin-diri

tags:
- secrets-management
- secrets-sprawl
- esc
- pulumi

social:
  twitter:
  linkedin:
---

There are several things in the lives of developers and DevOps engineers: taxes, yearly performance reviews, and secret sprawl. While the first two are inevitable, the last one is something that can be managed.

As we keep adding new cloud resources and releasing new applications, the number of secrets we need to manage keeps growing: passwords, API keys, certificates, and more. And as this is not enough, we need to manage secrets across different systems and environments with different teams that need to access them; we end up with duplicates.

> Let's have a look at a typical environment and see what teams use: Everything container-related may use Docker secrets or Kubernetes secrets. There is also a high chance that they might use cloud-specific secrets managers like Azure Key Vault, AWS Secrets Manager, or Google Secret Manager. In addition to these, there is a high chance that they might use HashiCorp Vault or some enterprise-grade secrets manager like CyberArk.

Author: I drop the example for now for a next blog post about this topic

---
title: "How Secrets Sprawl is Slowing You Down—And What to Do About It."
allow_long_title: true
date: 2024-10-28T12:40:10+02:00
draft: false

meta_desc: "Secrets sprawl is a real problem that can slow down our teams and impact our productivity. But it also affects how fast an organization can scale its development and operations teams. Learn how to overcome secrets sprawl and regain your productivity."

meta_image: meta.png

authors:
- engin-diri

tags:
- secrets-management
- secrets-sprawl
- esc
- pulumi

social:
  twitter: |
    Get rid of secrets sprawl and regain your productivity. 
    Learn how to overcome secrets sprawl and regain your productivity with Pulumi ESC. 
  linkedin: |
    Secrets sprawl is a real problem that can slow down our teams and impact our productivity. 
    But it also affects how fast an organization can scale its development and operations teams. 
    Learn how to overcome secrets sprawl and regain your productivity.
---

> There are several things in the lives of developers and DevOps engineers: taxes, yearly performance reviews, and secret sprawl. While the first two are inevitable, the last one is something that can be managed.

Reviewer: ```suggestion
Only a few things are certain in the lives of developers and DevOps engineers: taxes, yearly performance reviews, and secret sprawl. While the first two are inevitable, the last one is something that can be managed.
```
Changing to the english version of the idiom

> As we keep adding new cloud resources and releasing new applications, the number of secrets we need to manage keeps growing: passwords, API keys, certificates, and more. And as this is not enough, we need to manage secrets across different systems and environments with different teams that need to access them; we end up with duplicates.

Reviewer: ```suggestion
As we keep adding new cloud resources and releasing new applications, the number of secrets we need to manage keeps growing: passwords, API keys, certificates, and more. And as if this isn't enough, we need to manage secrets across different systems and environments with different teams that need to access them; we end up with duplicates.
```

Let's have a look at a typical environment and see what teams use: Everything container-related may use Docker secrets or Kubernetes secrets. There is also a high chance that they might use cloud-specific secrets managers like Azure Key Vault, AWS Secrets Manager, or Google Secret Manager. In addition to these, there is a high chance that they might use HashiCorp Vault or some enterprise-grade secrets manager like CyberArk.

> There are still a lot of organizations that store their secrets (like the API key for a cloud service) in environment variables to configure their applications on virtual machines, containers, or serverless functions. And not that this is a bad practice enough, there are also developers who still hardcode sensitive information in their application code. Or use plain unencrypted text files such as .env files and share them inside the team. To be fair, this is not something that is done on purpose, and most teams realize the risks of these practices. But the problem is that they have not found a better way to manage their secrets that won't slow them down and impact their productivity.

Reviewer: ```suggestion
There are still a lot of organizations that store their secrets (like the API key for a cloud service) in environment variables to configure their applications on virtual machines, containers, or serverless functions. And as if that weren’t risky enough, some developers still hardcode sensitive information in their application code. Or use plain unencrypted text files such as .env files and share them inside the team. To be fair, this is not something that is done on purpose, and most teams realize the risks of these practices. But the problem is that they have not found a better way to manage their secrets that won't slow them down and impact their productivity.
```
